pkgs/development/python-modules/qwt/default.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  numpy,
+  qtpy,
+  pyqt6,
+  qt6,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "qwt";
+  version = "0.12.6";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "PlotPyStack";
+    repo = "PythonQwt";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-SfedLlnq8mtAg5UeHkugmYCt6RnvnlRLdjBML+P6yVs=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    qtpy
+  ];
+  nativeCheckInputs = [
+    pytestCheckHook
+    # Not propagating this, to allow one to choose to either choose a pyqt /
+    # pyside implementation
+    pyqt6
```

Comment:
I agree and that's also not in upstream's dependencies.

Reply:
> I agree and that's also not in upstream's dependencies.

OTH the module is not functional without a qt python module available - this strengthens my position that simply imitating blindly upstream's `install_requires` is not always accurate - do you want to ensure a Python library is functional without any additional dependency? Or do you want to simply copy-paste lines from their `setup.py` or `pyproject.toml`?